### PR TITLE
STORY-11138: Add documentation for RingGroup and more

### PR DESCRIPTION
**Pull request by sstengele**
[STORY-11138](https://invoca.atlassian.net/browse/STORY-11138)

Adding documentation for RingGroup node and its params, as well as updating DynamicRoute for when it behaves like RingGroup, and accounting for Failover node as a child of Connect

## Screenshot Demo
Node Params and Usage
Node Details
Parameter Details

## Checklist

- [ ] Find the [Service owning team](https://docs.google.com/spreadsheets/d/1YF2wuepY5SZTpVEdT9gwhTPhpnDGZeVjywEwvbJS5TQ/edit#gid=0&fvid=1520238175) for these changes, and tag the team as "Reviewers" on this PR
- [ ] Test the documentation changes on readthedocs as a private branch
- [ ] If changing general content, have agreement on whether to apply to latest version or all versions (if all versions, provide links to the related PRs below)

### CODE REVIEW

File: source/api_documentation/network_integration/advertiser_campaigns/index.rst [hunk 1]
```diff
@@ -280,7 +280,14 @@ Node Parameters and Usage
...
 
-    - Forwards the call to a selected phone number after optionally reading a prompt.
+      connect_timeout
+
+    - Forwards the call to a selected phone number after optionally reading a prompt. If ringing_rollover is enabled for the network, a connect_timeout value can be configured along with a "Failover" child node.
+
+  * - Failover
+    - \*destination_phone_number
+
+    - Forwards the call to a fallback destination number if the original destination configured on the parent "Connect" node returns a busy signal OR if the configured connect_timeout value on the parent "Connect" node is exceeded. This node type can only be configured as a child of "Connect" nodes.
```
Comment (omeedrj) [inline source/api_documentation/network_integration/advertiser_campaigns/index.rst]: > This node type can only be configured as a child of "Connect" nodes.

This isn't necessarily accurate given that `ivr_ringing_rollover_extensions` has been made, right?

File: source/api_documentation/network_integration/advertiser_campaigns/index.rst [hunk 2]
```diff
@@ -403,6 +444,9 @@ Node Details
   * - YesOrNo
     - May have exactly 2 child nodes. If a keypress of 1 is made, the first child node is executed. If a kepyress of 2 is made, the second child node is executed. If speech recognition is enabled, the caller can also say "yes" for 1 and "no" for 2. At the end of the child list, this node type can also optionally have failover child nodes, designated by a node with a keypress_failover_type parameter (see example below).
 
+  * - RingGroup
+    - May not have any children. Forwards the call to a group of numbers that will be dialed in the order determined by the distribution method. If a destination plays a busy signal OR the ring_group_connect_timeout value is exceeded before the call is answered (or accepted), the next destination will be dialed. This node type also allows for simultaneous calling when it is enabled.
```
Comment (omeedrj) [inline source/api_documentation/network_integration/advertiser_campaigns/index.rst]: > May not have any children

Is this a little confusing given `ivr_ringing_rollover_extensions` functionality exists?

**Review by omeedrj — COMMENTED** [review]
Overall looks good, just 2 questions (which are the same)

File: source/api_documentation/network_integration/advertiser_campaigns/index.rst (hunk 1 above)
Comment (sstengele) [reply source/api_documentation/network_integration/advertiser_campaigns/index.rst]: Valid point, I just thought we were waiting to document `ivr_ringing_rollover_extensions`. I can remove the `"Connect"` reference and just have it say "parent node". I can also modify the last statment to say "This node type can only be configured as a child, and not as a standalone node." Thoughts?

Comment (omeedrj) [reply source/api_documentation/network_integration/advertiser_campaigns/index.rst]: Sounds good 👍

File: source/api_documentation/network_integration/advertiser_campaigns/index.rst (hunk 2 above)
Comment (sstengele) [reply source/api_documentation/network_integration/advertiser_campaigns/index.rst]: Valid point, but considering we're waiting to document `ivr_ringing_rollover_extensions`, saying it does allow a single child may be confusing as well. Thoughts?

Comment (omeedrj) [reply source/api_documentation/network_integration/advertiser_campaigns/index.rst]: You could just not reference "children" at all... 🤔 ?

What do you think?

**Review by omeedrj — APPROVED** [review]

**Pull request merged**